cpp/src/arrow/device.h

```diff
@@ -363,4 +363,33 @@ class ARROW_EXPORT CPUMemoryManager : public MemoryManager {
 ARROW_EXPORT
 std::shared_ptr<MemoryManager> default_cpu_memory_manager();
 
+using MemoryMapper =
+    std::function<Result<std::shared_ptr<MemoryManager>>(int64_t device_id)>;
```

Comment:
Sorry, but a couple more suggestions to unify naming:
1. rename `MemoryMapper` to `DeviceMemoryMapper`?
2. rename `RegisterDeviceMemoryManager` to `RegisterDeviceMemoryMapper`
3. rename `GetDeviceMemoryManager` to `GetDeviceMemoryMapper`

Reply:
Good points, that naming is definitely more consistent. 

There is however one problem that we already define a `DeviceMemoryMapper` for the keyword type in the actual bridge.h Import methods:

https://github.com/apache/arrow/blob/434f87274e8e9adab4f0434ae494f30dc955ca6e/cpp/src/arrow/c/bridge.h#L218-L219

and we should probably find a distinct name, given that both are slight different (the one takes device_type+device_id and returns a MemoryManager, while the other is a function already tied to a specific device_type and thus only takes a device_id, returning again a MemoryManager)

It's of course a subtle difference that might be difficult to embody in a name. But at least using distinct names seems best.

Reply:
Perhaps `DeviceIdMapper` then? Not terribly pretty I admit...

Reply:
I think that sounds good for the function type alias, but then I would personally leave the register/get functions as is? I would find `RegisterDeviceIdMapper` a bit strange with the focus on the `id`, because you are also registering a device _type_, it's just that the value you store for the registered _type_ is the DeviceIdMapper ..

Anyway, in the end it doesn't matter that much, happy to go with whatever we come up with.

Reply:
Or `DeviceMapper` / `RegisterDeviceMapper` / `GetDeviceMapper` ?  (that's a bit more generic, but keeps the three consistent with each other)